content/posts/git-worktree-workflow/index.md

```diff
@@ -0,0 +1,455 @@
+---
+title: "Git Worktrees + DataLad: The Missing Link Between Daily Development and Reproducible Analysis"
+date: 2025-12-08T19:42:00+02:00
+author:
+- Jiameng Wu
+tags:
+- Datalad
+- git-worktree
+- YODA
+- Workflow
+- data-analysis
+cover:
+  image: cover.jpeg
+  alt: 
+  relative: true
+description: >
+  Bridging fast iteration during development and reproducible batch processing with git worktrees and DataLad. 
+  A practical workflow that is suitable for the 90% of work that lies between the initial exploration and the final publication.
+showToc: true
+# hide top-level metadata (date, length, author)?
+hidemeta: false
+# no highlighting of "code"
+disableHLJS: true # to disable highlightjs
+# no sharing buttons
+disableShare: false
+# no summary shown at the top
+hideSummary: false
+---
+
+In neuroscience research, we often find ourselves caught between two worlds: the messy, exploratory world of data analysis where we iterate rapidly on code, and the structured, reproducible world of scientific publication where every figure must be exactly reproducible. Between these worlds lies a labyrinth of processing pipelines, half-written scripts, and the ever-present risk of breaking your working analysis while trying to improve it.
+
+Today's solutions often fall into two extremes: heavyweight reproducible workflows like [fairly-big-processing-workflow](https://github.com/psychoinformatics-de/fairly-big-processing-workflow) that are beautiful for final analysis but feel like overkill for daily development, or ad-hoc scripts that get the job done but leave no trace of how they did it.
+
+At the recent [Distribits](https://www.distribits.live/events/2025-distribits/) meeting, I shared my struggle with [Yarik](https://github.com/yarikoptic), and confessed that I never fully understood the [YODA principle](https://handbook.datalad.org/en/latest/basics/101-127-yoda.html) - I was on the way, but I couldn't see clearly where I was heading. Together, we've found a sweet spot that gives me both **fast iteration during development** and **clean reproducibility for batch processing** - all without duplicating data or rebuilding containers. The secret? [Git worktrees](https://git-scm.com/docs/git-worktree) combined with [DataLad](https://handbook.datalad.org/en/latest/index.html)'s nested datasets.
+
+### When YODA's wisdom grows on trees
+```
```

Comment:
Could you also add a sentence or two on the opening on what your study is about and what you are aiming here at (preprocessing vs eg paper figures)